Makefile

```diff
@@ -22,24 +22,24 @@ debug:
 
 perf-release:
 	CARGO_PROFILE_RELEASE_DEBUG=true cargo $(RUST_OPTIONS) build -p neard --release --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p near-vm-runner-standalone --release --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p state-viewer --release --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p store-validator --release --features performance_stats,memory_stats
+	cargo $(RUST_OPTIONS) build -p near-vm-runner-standalone --release
+	cargo $(RUST_OPTIONS) build -p state-viewer --release --features nearcore/performance_stats,nearcore/memory_stats
+	cargo $(RUST_OPTIONS) build -p store-validator --release --features nearcore/performance_stats,nearcore/memory_stats
 
 perf-debug:
 	CARGO_PROFILE_RELEASE_DEBUG=true cargo $(RUST_OPTIONS) build -p neard --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p near-vm-runner-standalone --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p state-viewer --features performance_stats,memory_stats
-	cargo $(RUST_OPTIONS) build -p store-validator --features performance_stats,memory_stats
+	cargo $(RUST_OPTIONS) build -p near-vm-runner-standalone
+	cargo $(RUST_OPTIONS) build -p state-viewer --features nearcore/performance_stats,nearcore/memory_stats
+	cargo $(RUST_OPTIONS) build -p store-validator --features nearcore/performance_stats,nearcore/memory_stats
```

Comment:
we probably should check these build in .buildkite/pipeline.yml:
```
cargo check -p state-viewer --features nearcore/performance_stats,nearcore/memory_stats
cargo check -p store-validator --features nearcore/performance_stats,nearcore/memory_stats
```
because existing check doesn't check these feature combinations

Reply:
Aren't these covered by this? https://github.com/near/nearcore/blob/928f15df25ca4128d88b98e930d580ed2023d576/.buildkite/pipeline.yml#L30
And @chefsale, what do you think? Regarding your remark in https://github.com/near/nearcore/pull/4338#issuecomment-850899095, if those lines aren't necessary, checks for them might not as well.

Reply:
No it doesn't because there's no workspace features, see this two commits' buildkite:
https://github.com/near/nearcore/pull/4287/commits/1a1256ce65cfc6a0076da19ecbeb16790ce214df
https://github.com/near/nearcore/pull/4287/commits/08e7e31e2482850f515acaacb8d04a8d30b4a74b

So the first one only checks
```
       RUSTFLAGS='-D warnings' cargo check --workspace --all-targets --all-features 
```
It passed. The second commit add a `cargo check -p neard --features sandbox` feature and it failed. This means sandbox feature of neard was not checked by ` cargo check --workspace --all-targets --all-features `

Reply:
@miraclx I think you're correct, all-features do include all features, mine case was failing because it's missing some features instead of `--all-features` didn't work. So the ci doesn't check all feature combinations. `cargo check --workspace --all-targets --all-features` ensures when all features are included, all packages can compile. But real build includes package build with no feature, or a subset of features, and they can fail if feature dependency wasn't correctly specified.

So for example this one from Makefile
```
cargo build -p state-viewer --features nearcore/performance_stats,nearcore/memory_stats
```
since we don't do `make` on ci, it makes sense to cargo check these compile, not introduced by your PR, but a thing we probably should do to gurantee `make` works on master commits.